osu.Game/Online/Chat/Channel.cs

```diff
@@ -51,11 +54,18 @@ public void AddLocalEcho(LocalEchoMessage message)
             NewMessagesArrived?.Invoke(new[] { message });
         }
 
+        public bool MessagesLoaded { get; private set; }
+
         public void AddNewMessages(params Message[] messages)
         {
             messages = messages.Except(Messages).ToArray();
 
             Messages.AddRange(messages);
+            MessagesLoaded = true;
+
+            var maxMessageId = messages.Max(m => m.Id);
```

Comment:
Wouldn't this fail if ``messages`` is empty`?

Reply:
It will, yes. In all current usages that is not possible, but I agree that adding safety here is probably best.